src/org/labkey/targetedms/outliers/OutlierGenerator.java

```diff
@@ -135,14 +152,27 @@ private String queryContainerSampleFileRawData(List<QCMetricConfiguration> confi
         sql.append("\nON sf.ReplicateId = exclusion.ReplicateId AND (exclusion.MetricId IS NULL OR exclusion.MetricId = x.MetricId)");
         sql.append("\nLEFT JOIN GuideSetForOutliers gs");
         sql.append("\nON ((sf.AcquiredTime >= gs.TrainingStart AND sf.AcquiredTime < gs.ReferenceEnd) OR (sf.AcquiredTime >= gs.TrainingStart AND gs.ReferenceEnd IS NULL))");
-        if (null != startDate && null != endDate)
+        if (null != startDate || null != endDate)
         {
-            sql.append("\nWHERE sf.AcquiredTime >= '");
-            sql.append(startDate);
-            sql.append("' AND ");
-            sql.append("\n sf.AcquiredTime < TIMESTAMPADD('SQL_TSI_DAY', 1, CAST('");
-            sql.append(endDate);
-            sql.append("' AS TIMESTAMP))");
+            sql.append("\nWHERE ");
+
+            if (null != startDate)
+            {
+                sql.append("sf.AcquiredTime >= '");
+                sql.append(startDate);
+                sql.append("' AND ");
+            }
+
+            if (null != endDate)
+            {
+                sql.append("\n sf.AcquiredTime < TIMESTAMPADD('SQL_TSI_DAY', 1, CAST('");
+                sql.append(endDate);
+                sql.append("' AS TIMESTAMP))");
+            }
+            else
+            {
+                sql.append("1=1");
```

Comment:
Perhaps cleaner to use a separator variable that's first set to "WHERE" and then to "AND", to avoid the 1=1 in the else